From a2a6d2e4aafc3bed082a3d21e2c470492ef468b1 Mon Sep 17 00:00:00 2001
From: Johannes Maibaum <jmaibaum@gmail.com>
Date: Sun, 11 Apr 2021 13:19:02 +0200
Subject: [PATCH] weather-service: Set contact_info to DOAP file URL

Currently, GNOME Calendar does not set up the contact_info needed in
libgweather 40 API. This leads to the following assertion, and no
weather forecast info being shown in Calendar:

`gweather_info_update: assertion 'info->contact_info != NULL' failed.`

To fix the assertion, set the URL to the DOAP file on the master branch.
GNOME Weather does the same, see:

https://gitlab.gnome.org/GNOME/gnome-weather/-/commit/6763f7dc9cb25eac4aa13e94e07ea932a42c8e4d

Part of #682
---
 src/weather/gcal-weather-service.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/weather/gcal-weather-service.c b/src/weather/gcal-weather-service.c
index 4cf47f02..f67afd69 100644
--- a/src/weather/gcal-weather-service.c
+++ b/src/weather/gcal-weather-service.c
@@ -637,6 +637,8 @@ update_location (GcalWeatherService  *self,
                !location ? "<null>" : gweather_location_get_name (location));
 
       self->gweather_info = gweather_info_new (location);
+      gweather_info_set_contact_info (self->gweather_info,
+                                      "https://gitlab.gnome.org/GNOME/gnome-calendar/-/raw/master/gnome-calendar.doap");
 
       /* FIXME: Enable GWEATHER_PROVIDER_MET_NO if we can comply with their terms of service.
        * https://gitlab.gnome.org/GNOME/gnome-calendar/-/issues/682
-- 
GitLab

